src/main/java/io/harness/cf/client/api/AuthService.java

```diff
@@ -33,10 +33,16 @@ protected void runOneIteration() {
       stopAsync();
       log.info("Stopping Auth service");
     } catch (ConnectorException e) {
-      log.error(
-          "Exception while authenticating, retry in {} seconds, error: {}",
-          pollIntervalInSec,
-          e.getMessage());
+      if (e.shouldRetry()) {
```

Comment:
Added a `shouldRetry` to the `ConnectorException` so we can indicate situations that won't improve with a retry (bad api key, for example)